src/search-jobs.js

```diff
@@ -2,87 +2,110 @@ const Apify = require('apify');
-            }
-            rawdata = $('li.react-job-listing');
-            json = rawdata
-                .map(mapJobListItem)
-                .get();
-        } catch (error) {
-            if (error.statusCode === 504 && limitRetries < 5) {
-                log.info(' - Encountered rate limit, waiting 3 seconds');
-                await Apify.utils.sleep(3000);
-                limitRetries++;
-                continue; // eslint-disable-line
-            } else {
-                // Rethrow non rate-limit errors or if we are stuck
-                throw error;
+    if (savedItems === 0 || savedItems < maximumResults) {
+        // REQUEST ITSELF
+        const rq = await Apify.utils.requestAsBrowser({
+            url: request.url,
+            proxyUrl: proxyConfiguration.newUrl(session.id),
+        });
+        // ADDING CHEERIO
+        $ = cheerio.load(rq.body);
+        // IF THERE IS NO MAX RESULT IN THE INPUT => GETTING ALL AVAILABLE
+        if (maximumResults < 0) {
+            // COUNTER OF RESULTS ON THE PAGE
+            const cntStr = $('p[data-test="jobsCount"]').text().replace(',', '');
+            maximumResults = parseInt(cntStr, 10);
+            if (!(maximumResults > 0)) {
+                throw new Error(`Failed to parse jobsCount from ${cntStr}`);
             }
+            log.info(`Parsed maximumResults = ${maximumResults}`);
         }
-        console.dir(json);
+        const rawdata = $('li.react-job-listing');
+        const json = rawdata
+            .map(mapJobListItem)
+            .get();
         itemsToSave = json.slice(0, maximumResults - savedItems);
         searchResults.push(...itemsToSave);
         savedItems += itemsToSave.length;
-        nextPageUrl = $('a[data-test="pagination-next"]', '#FooterPageNav').attr('href');
-        log.info(`Page ${page}: Found ${itemsToSave.length} items, next page: ${nextPageUrl}`);
-        page++;
-    } while (nextPageUrl && savedItems < maximumResults && itemsToSave && itemsToSave.length > 0);
 
-    return searchResults;
+        let nextPage;
+        const currentUrl = request.url;
+        if (page === 1) {
+            // FOR THE NEXT PAGES THERE IS DIFFERENT STRUCTURE OF THE URL - GETTING IT FROM META TAG ON THE PAGE
+            nextPage = $('meta[property="og:url"]').attr('content');
+            // PAGINATION IS REPRESENTED BY '_IP2,3,4,5...'
+            nextPage = nextPage.replace('.htm', '_IP2.htm');
+        } else if (page > 1) {
+            nextPage = currentUrl.replace(currentUrl.match(/IP([0-9.]+)/)[1], `${page}.`);
+        }
+        // SECOND PAGE WAS ADDED ABOVE => ADDING THIRD ONE
+        if (page === 1) {
+            page += 2;
+        } else {
+            page += 1;
+        }
+        currentPage = +$('div[data-test="page-x-of-y"]').text().replace('Page', '').split('of')[0].trim();
+        maxPage = +$('div[data-test="page-x-of-y"]').text().replace('Page', '').split('of')[1].trim();
```

Comment:
this is randomly failing with `TypeError: Cannot read property 'trim' of undefined`. need to see in which situations this is happening, because it will break the pagination